frontend/src/app/spindle/spindle.component.ts

```diff
@@ -39,41 +58,61 @@ export class SpindleComponent implements OnInit, OnDestroy {
                     this.errorHandler.handleSpindleError(response.error);
                     return;
                 }
-                if (response.tex) {
-                    this.texOutput = response.tex;
+                if (response.latex) {
+                    this.textOutput = {
+                        extension: 'tex',
+                        text: response.latex
+                    }
                 }
                 if (response.redirect) {
                     // Opens a new tab.
                     window.open(response.redirect, "_blank");
                 }
                 if (response.pdf) {
-                    this.downloadPDF(response.pdf);
+                    const base64 = response.pdf;
+                    const linkSource = `data:application/pdf;base64,${base64}`;
+                    const fileName = "spindleParseResult.pdf";
+                    this.downloadFile(fileName, linkSource);
+                }
+                if (response.term && response.lexical_phrases) {
+                    this.term = response.term;
+                    this.lexicalPhrases = response.lexical_phrases;
+                }
+                if (response.proof) {
+                    this.textOutput = {
+                        extension: 'json',
+                        text: this.JSONPipe.transform(response.proof)
```

Comment:
Is using `JSONPipe` different than `JSON.stringify()`? If it is, maybe add a comment explaining it

Reply:
Angular's `JSONPipe` adds newlines and spaces so the resulting string can be pretty-printed. However, I learned that `JSON.stringify` can do the same if you pass additional arguments.